packages/worker/src/debug-replayer/client.ts

```diff
@@ -41,10 +41,19 @@ export class Client {
     }
     const response = await new Promise<http.IncomingMessage>((resolve, reject) => {
       request.once('error', reject);
-      request.once('response', resolve);
+      request.once('response', (res) => {
+        let body = [];
+        res.on('data', function (chunk) {
+          body.push(chunk);
+        });
+        res.on('end', function () {
+          res.body = JSON.parse(body.toString());
+          resolve(res);
+        });
+      });
```

Comment:
You can use the response as an async iterator if you need the body.
In some cases we don't want to consume the body as JSON, I would only consume it on error.

Reply:
That makes total sense. Updated ✅